src/igmp.c

```diff
@@ -136,39 +139,42 @@ void acceptIgmp(int recvlen) {
-                    checkVIF = getIfByIx( upStreamIfIdx[i] );
-                    if(checkVIF == 0) {
+            for (i = 0; i < MAX_UPS_VIFS; i++) {
+                if (-1 != upStreamIfIdx[i]) {
+                    // Check if the source address matches a valid address on upstream
+                    // vif.
+                    checkVIF = getIfByIx(upStreamIfIdx[i]);
+                    if (checkVIF == 0) {
                         my_log(LOG_ERR, 0, "Upstream VIF was null.");
                         return;
-                    }
-                    else if(src == checkVIF->InAdr.s_addr) {
-                        my_log(LOG_NOTICE, 0, "Route activation request from %s for %s is from myself. Ignoring.",
-                            inetFmt(src, s1), inetFmt(dst, s2));
+                    } else if (src == checkVIF->InAdr.s_addr) {
+                        my_log(LOG_NOTICE, 0,
+                               "Route activation request from %s for %s is "
+                               "from myself. Ignoring.",
+                               inetFmt(src, s1), inetFmt(dst, s2));
                         return;
-                    }
-                    else if(!isAdressValidForIf(checkVIF, src)) {
+                    } else if (!isAdressValidForIf(checkVIF, src)) {
                         struct IfDesc *downVIF = getIfByAddress(src);
                         if (downVIF && downVIF->state & IF_STATE_DOWNSTREAM) {
-                            my_log(LOG_NOTICE, 0, "The source address %s for group %s is from downstream VIF[%d]. Ignoring.",
-                                inetFmt(src, s1), inetFmt(dst, s2), i);
+                            my_log(LOG_NOTICE, 0,
+                                   "The source address %s for group %s is "
+                                   "from downstream VIF[%d]. Ignoring.",
+                                   inetFmt(src, s1), inetFmt(dst, s2), i);
                         } else {
-                            my_log(LOG_WARNING, 0, "The source address %s for group %s, is not in any valid net for upstream VIF[%d].",
-                                inetFmt(src, s1), inetFmt(dst, s2), i);
+                            my_log(LOG_WARNING, 0,
+                                   "The source address %s for group %s, is "
+                                   "not in any valid net for upstream "
+                                   "VIF[%d].",
+                                   inetFmt(src, s1), inetFmt(dst, s2), i);
                         }
                     } else {
                         // Activate the route.
                         int vifindex = checkVIF->index;
-                        my_log(LOG_DEBUG, 0, "Route activate request from %s to %s on VIF[%d]",
-                            inetFmt(src,s1), inetFmt(dst,s2), vifindex);
+                        my_log(LOG_DEBUG, 0, "Route activate request from %s to %s on VIF[%d]", inetFmt(src, s1),
```

Comment:
This is the result of automatic reformatting.

It can be changed by having other formatting options or  
it could be solved by using clang directives such as

    int formatted_code;
    // clang-format off
        void    unformatted_code   ;
    // clang-format on
    void formatted_code_again;

I would prefer rather not introduce clang specific stuff into the code.